straxen/plugins/acqmon_processing.py

```diff
@@ -216,3 +217,138 @@ def compute(self, events, aqmon_hits):
+                 help=('Internal delay between aqmon and regular TPC channels ins [ns]'
+                       'before/after run: 21286')
+                 ),
+    strax.Option('adc_threshold_nim_signal', default=500, type=int, track=True,
+                 help='Threshold in [adc] to search for the NIM signal'
+                 ),
+    strax.Option('epsilon_offset', default=0, type=(int, float), track=True,
+                 help='Measured missing offset for nveto in [ns]'
+                 ),
+)
+class DetectorSynchronization(strax.Plugin):
+    """
+    Plugin which computes the synchronization delay between TPC and
+    vetos.
+
+    Reference:
+        * xenon:xenonnt:dsg:mveto:sync_monitor
+    """
+    __version__ = '0.0.1'
+    depends_on = ('raw_records_aqmon',
+                  'raw_records_aqmon_nv',
+                  'raw_records_aux_mv')
+    provides = 'detector_time_offsets'
+    data_kind = 'detector_time_offsets'
+
+    def infer_dtype(self):
+        dtype = []
+        dtype += strax.time_fields
+        dtype += [((('Time offset for nV to synchornize with TPC in [ns]'),
+                    'time_offset_nv'), np.int64),
+                  ((('Time offset for mV to synchornize with TPC in [ns]'),
+                    'time_offset_mv'), np.int64)
+                  ]
+        return dtype
+
+    def compute(self, raw_records_aqmon,
+                raw_records_aqmon_nv,
+                raw_records_aux_mv,
+                start, end):
+        rr_tpc = raw_records_aqmon
+        rr_nv = raw_records_aqmon_nv
+        rr_mv = raw_records_aux_mv
+
+        extra_offset = 0
+        _mask_tpc = (rr_tpc['channel'] == 798)
+        if not np.any(_mask_tpc):
+            # For some runs in the beginning no signal has been acquired here.
+            # In that case we have to add the internal DAQ delay as an extra offset later.
+            _mask_tpc = (rr_tpc['channel'] == 801)
+            extra_offset = self.get_delay()
+
+        hits_tpc = self.get_nim_edge(rr_tpc[_mask_tpc], self.config['adc_threshold_nim_signal'])
+
+        _mask_mveto = (rr_mv['channel'] == 1084)
+        hits_mv = self.get_nim_edge(rr_mv[_mask_mveto], self.config['adc_threshold_nim_signal'])
+
+        _mask_nveto = rr_nv['channel'] == 813
+        hits_nv = self.get_nim_edge(rr_nv[_mask_nveto], self.config['adc_threshold_nim_signal'])
+        nveto_extra_offset = 0
+        if not len(hits_nv):
+            # During SR0 sync signal was not recorded properly for the
+            # neutron-veto, hence take waveform itself as "hits".
+            _mask_nveto &= rr_nv['record_i'] == 0
+            nveto_extra_offset = self.config['epsilon_offset']
+            hits_nv = rr_nv[_mask_nveto]
+
+        offsets_mv = self.estimate_delay(hits_tpc, hits_mv)
+        offsets_nv = self.estimate_delay(hits_tpc, hits_nv)
+        assert len(offsets_mv) == len(offsets_nv), 'Unequal number of sync signals!'
```

Comment:
They should be the same length, but how do we manage if this condition is not met? In principle we can calculate the offsets even if mv or nv is missing one of the sync signals for the run.

Reply:
I changed the logic a bit. In case we are missing one signal we fill the corresponding row with the error value. Hence, the two offsets should always have the same length unless there is bug in the code.